mobilecoind/src/service.rs

```diff
@@ -537,11 +537,32 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
             rpc_internal_error("ledger_db.get_tx_out_by_index", err, &self.logger)
         })?;
 
-        // Use root entropy to construct AccountKey.
-        let mut root_entropy = [0u8; 32];
-        root_entropy.copy_from_slice(transfer_payload.get_root_entropy());
-        let root_id = RootIdentity::from(&root_entropy);
-        let account_key = AccountKey::from(&root_id);
+        // Use bip39 or root entropy to construct AccountKey.
+        let account_key = if !transfer_payload.get_bip39_entropy().is_empty() {
+            let mut bip39_entropy = [0u8; 32];
+            if bip39_entropy.len() != transfer_payload.get_bip39_entropy().len() {
```

Comment:
Are we wanting to only support 32 byte bip-39 entropies? I don't think I've seen this restriction anywhere else so far.

Reply:
I'll remove the limitation